src/models/chan.js

```diff
@@ -129,7 +129,7 @@ Chan.prototype.getMode = function(name) {
 Chan.prototype.toJSON = function() {
 	var clone = _.clone(this);
 	clone.users = []; // Do not send user list, the client will explicitly request it when needed
-	clone.messages = clone.messages.slice(-100);
```

Comment:
could lowering this be a better solution?

Reply:
Lowering the number of messages would postpone the issue until a user has more channels, but it wouldn't actually _solve_ the problem. Having the client fetch messages when a channel is activated, just like it does with user lists, provides the simplest solution to having a big fat blob of init data that has to be parsed all in one go. Lazy-loading messages for inactive channels in the background can come later, right @xPaw?

Reply:
I could probably send one message for each chananel (that'll fix the possible `lastMessageId` issue). Also I think there is no need to lazily send messages after login.